src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.tsx

```diff
@@ -157,13 +166,17 @@ function ContactMethodDetailsPage({route}: ContactMethodDetailsPageProps) {
         setIsValidateCodeActionModalVisible(!loginData?.validatedDate);
     }, [loginData?.validatedDate, loginData?.errorFields?.addedLogin]);
 
+    useEffect(() => {
+        resetContactMethodValidateCodeSentState(contactMethod);
+    }, [contactMethod]);
```

Comment:
This caused the `loginData.validatedDate` to briefly turn `undefined` even if the user is actually verified. This triggered the above navigation effect which made it impossible to enter this page. https://github.com/Expensify/App/issues/56355

We should have called `resetContactMethodValidateCodeSentState` only for non verified contacts.